spec/services/image_filename_service_spec.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+RSpec.describe ImageFilenameService do
+  describe "#call" do
+    it "parameterises the base filename" do
+      document = build :document
+      name = ImageFilenameService.new(document).call("File $ name.jpg", "image/jpeg")
+      expect(name).to eq "file-name.jpg"
+    end
+
+    it "enforces an extension for the filename" do
+      document = build :document
+      name = ImageFilenameService.new(document).call("file.jpg", "image/png")
+      expect(name).to eq "file.png"
+    end
+
+    it "ensures the filename is unique" do
+      document = create(:document)
+      create(:image, document: document, filename: "name.jpg")
+      create(:image, document: document, filename: "name-1.jpg")
+      create(:image, filename: "name-2.jpg")
```

Comment:
I think you can remove this line. Unless you change the test name to be "ensures filenames of images associated with a document are unique"?